### DISCUSSION TITLE
Use of NTP fallback when NTP servers are configured ?

### DISCUSSION THREAD

**Discussion by FulvioSpelta** [arendst/Tasmota#20978] [Q&A]
Hi all,
I'm using an WT32-ETH01 configured with:

- wifi off
- EthIpAddress, EthSubnetmask, EthGateway; EthDNSServer1; EthDNSServer2; fixed
- Backlog NtpServer1 10.193.8.10; NtpServer2 ""; NtpServer3 "";
note: the NtpServer1 is the EthGateway too

From the console log it seems that tasmota still use 0.pool.ntp.org as ntp server, log:
```
10:35:39.348 NTP: Sync time...
10:35:39.371 WIF: DNS resolved '0.pool.ntp.org' (93.94.88.50) in 20 ms
10:35:39.405 RTC: UTC 2024-03-16T09:35:39Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
10:35:39.000 RTC: Synced by NTP
```

log of command to check the ntp configuration:
```
10:49:52.734 CMD: ntpserver1
10:49:52.736 SRC: WebConsole from 192.168.10.105
10:49:52.738 CMD: Grp 0, Cmd 'NTPSERVER', Idx 1, Len 0, Pld -99, Data ''
10:49:52.743 MQT: smartbox/stat/smartbox-002/RESULT = {"NtpServer1":"10.193.8.10"}
10:49:55.838 BRY: GC from 22978 to 12531 bytes, objects freed 163/182 (in 2 ms) - slots from 433/470 to 166/335
10:49:58.186 CMD: ntpserver2
10:49:58.188 SRC: WebConsole from 192.168.10.105
10:49:58.190 CMD: Grp 0, Cmd 'NTPSERVER', Idx 2, Len 0, Pld -99, Data ''
10:49:58.193 MQT: smartbox/stat/smartbox-002/RESULT = {"NtpServer2":""}
10:50:01.839 CMD: ntpserver3
10:50:01.841 SRC: WebConsole from 192.168.10.105
10:50:01.843 CMD: Grp 0, Cmd 'NTPSERVER', Idx 3, Len 0, Pld -99, Data ''
10:50:01.846 MQT: smartbox/stat/smartbox-002/RESULT = {"NtpServer3":""}
```

I'm wrong on something ?
Thanks

**Comment by sfromis** [comment]
You can't configure NTP servers away, there will always be a built-in fallback. What you can do is to give Tasmota another real time source, like a RTC or GPS chip. Additionally, using the `Time` command to set a time will disable NTP.

I'd suggest having a NTP server on your network, if you want no external NTP server access. Many options for having a simple NTP server, this is a feature of many routers or NAS boxes. Even Tasmota can act as a NTP server, of course with a good time source.

**Reply by FulvioSpelta** [reply]
Thanks for your comment. 
My configuration is exactly what you are suggesting: The NTP server is hosted on a device reachable on the network.

To be clearer i post the IP conf:
```
Backlog EthIpAddress 192.168.10.69; EthSubnetmask 255.255.255.0; EthGateway 192.168.10.1; EthDNSServer1 8.8.8.8; EthDNSServer2 8.8.4.4;
Backlog NtpServer1 10.193.8.10; NtpServer2 ""; NtpServer3 "";
```

The question is: have I to change the conf to avoid the ntp fallback ?
Thanks

Is OT but just to share: i use the TIME command this way:
At startup i use the rule
`rule2 on power1#boot do time 1704067200 endon;` (set time to 1 January 2024 00:00:00)
in order to enable tasmota to run the schedule functions even without ntp sync

After 1 min, using berry, i issue the command:
`time 0`
to enable the ntp sync as soon as possible maintaining the schedule capability of tasmota.

**Reply by sfromis** [reply]
Again, you can't configure the fallback away. I already told you the options.

And you can't guarantee the fallback not being used if your NTP server is not reachable. Doing the `Time 0` later, after already taking the detrimental effects from using a fake time, does not preclude use of the fallback. Anyway, if you are not providing a DNS resolution of the NTP server names, they will not be reachable.
